pages/overview.mdx

```diff
@@ -1,29 +1,111 @@
 # Technical Overview
+## What happens when you publish a site to Distributed Press?
+When you [upload content to a site](./api-reference/site#uploading-content-to-a-site) on Distributed Press, the server first 
+untars the files to a temporary folder on the server.
 
-Distributed Press has many components that work together.
+Then, the server looks up the associated configuration for the site. It figures out what 
+protocols the user has indicated interest in publishing to and then kicks off sync tasks
+for the enabled protocols.
 
-## API Server
-The API server is how clients interface with Distributed Press. 
+### IPFS
+For IPFS, the associated logic is [in this file](https://github.com/hyphacoop/api.distributed.press/blob/main/protocols/ipfs.ts).
+The main things you need to know is that Distributed Press keeps a long-running [Kubo](https://www.npmjs.com/package/go-ipfs)
+instance that is spun-up when Distributed Press starts up. This connection is managed by `js-ipfsd-ctl`.
 
-Clients make authenticated HTTP requests to make updates to resources 
-like publishers, admins, and sites.
+We then copy the files to an IPFS [Mutable File System (MFS)](https://docs.ipfs.tech/concepts/file-systems/) and 
+publish it to IPFS. When finished, this returns us a CID. We update the IPNS entry for the site ID to this resulting CID.
 
-The API server is responsible for locally storing site data to serve and 
-seed over the various [peer-to-peer protocols](./protocols) that Distributed Press supports.
+This CID and accompanying IPNS entry is used to populate the `links` section:
 
-See the [API reference](./api-reference/) for more details.
+```typescript
+type IPFSLinks = {
+  "gateway": string,
+  "cid": string,
+  "pubKey": string,
+  "dnslink": string,
+  "enabled": boolean,
+  "link": string 
+}
+```
 
+### Holepunch
+The associated logic for Holepunch is [in this file](https://github.com/hyphacoop/api.distributed.press/blob/main/protocols/hyper.ts).
+Holepunch similarly uses [`localdrive`](https://www.npmjs.com/package/localdrive) which overs a mutable file system API that is similar to Hyperdrive.
 
-## DNS
-The DNS server runs alongside the Distributed Press API and is responsible
-for resolving DNS queries for sites that it hosts.
+We create a drive for each site and then copy the files into it. This is done through `localdrive` as it has a feature to
+mirror files from your filesystem to the drive.
 
-This means that if you own the domain for the site you host (e.g. `distributed.press`),
-you can add an `NS` record through your domain registrar to tell it to look at
-Distributed Press for where to find where the site is actually hosted.
+When finished, we get the drive URL and similarly populate the `links` section:
```

Comment:
Where can once access the HTTP links and CIDs that one has published to? 

For example, with Github Actions, this is output in the Actions workflow. Not sure how Sutty shares this, but would be good to reference where one can find this

Reply:
<img width="1328" alt="dpress-actions-output" src="https://github.com/hyphacoop/docs.distributed.press/assets/66978602/d93a23fa-7acd-4ece-a79d-8f6052c2e5a5">